app.js

```diff
@@ -1,17 +1,36 @@
-import { stdin as input, stdout as output } from 'node:process';
-import * as readline from 'node:readline/promises';
+import { WordWrapDecorator } from './src/decorators.js';
+import { scenario } from './src/scenario.js';
+import { negativeFinishStrategy, positiveFinishStrategy, promptStrategy, Scene } from './src/scene.js';
+import { State } from './src/State.js';
 
-const rl = readline.createInterface({ input, output });
+let state = new State(scenario);
 
-rl.on('line', async (line) => {
-    try {
-        console.log(line);
-    } catch {
-        console.log('Что-то пошло не так. Игра окончена!');
-        rl.close();
+
+const playScene = async () => {
+    const currentLevel = state.getCurrentLevel();
```

Comment:
it's a great idea to use state here. But instead of using `ifs` it would be great to move this logic to the State class itself. Then it'll represent different states and you can easily process state with some single function call